### ISSUE TITLE
`router` fails if URL has hash (#) fragment

### ISSUE THREAD

**Issue by gapmiss** [nuejs/nue#161]
Testing out `nuemark` and having issues with the page router.

For example:

A page with a nuemark tabs component
http://localhost:8080/index.html

click a tab and the url changes to
http://localhost:8080/index.html#tab-1

click link to a different page
http://localhost:8080/example.html

click browser's back button and the URL is again
http://localhost:8080/index.html#tab-1

however, the page content is still from the `example.html`

screencast:

![CleanShot-Firefox-Markdown with megapowers-2024-01-13-09 07 13](https://github.com/nuejs/nue/assets/98914514/560cab53-eb7d-40a9-8c39-b45b4de3033f)

**Comment by tipiirai** [comment]
Ahh... good catch. I forgot to implement the collaboration between the tabs component and the page-router. Thanks!

**Comment by fritx** [comment]
Forgive me I might be off-topic, but I'm borrowing the topic of "page_router":

Is the `page_router` in site.yaml now deprecated in favor of `router: true|false`?
I'm seeing only `if (data.router) push('page_router')`...

But there are still two places that using `page_router: true` in create-www

**Comment by nobkd** [comment]
@fritx yes, it has been replaced. See #113 with its closing commit https://github.com/nuejs/nue/commit/be2f19ae0dc564759570c38771ab47bb41f35f63. Then this should be corrected in create and www

**Comment by nobkd** [comment]
I think tabs don't use the URL hash (`#`) anymore, and therefore shouldn't involve navigation history.

Please reopen, if closing this was somehow incorrect. Thank you.

**Comment by nobkd** [comment]
Wrongly closed. Sry.
